src/utils.jl

```diff
@@ -338,3 +338,49 @@ function show_stackloc(io::IO, frame)
     end
 end
 show_stackloc(frame) = show_stackloc(stdout, frame)
+
+# Pretty printing of stacktraces and errors with Frame
+function Base.StackTraces.StackFrame(frame::Frame)
+    if frame.framecode.scope isa Method
+        method = frame.framecode.scope
+        method_args = something.(frame.framedata.locals[1:frame.framecode.scope.nargs])
+        atypes = Tuple{typeof.(method_args)...}
+        sig = frame.framecode.scope.sig
+        (ti, lenv) = ccall(:jl_type_intersection_with_env, Any, (Any, Any), atypes, sig)
+        mi = Core.Compiler.code_for_method(method, atypes, lenv, typemax(UInt))
+    else
+        mi = frame.framecode.src
```

Comment:
Is the CodeInfo what normally gets inserted when toplevel throws an error? I don't know the answer myself. If you haven't tested you can use `prepare_thunk` and see.

Reply:
Looking at the printing, that seems to be the case:

```
julia> foo(x) = x
foo (generic function with 1 method)

julia> Base.StackTraces.StackFrame(:foo, Symbol("file.jl"), 5, (@code_lowered foo(2)), false, false, C_NULL)
top-level scope at file.jl:5
```